packages/client/auth-core/src/utils/jwt.ts

```diff
@@ -1,25 +1,59 @@
 import { CrossmintService } from "@/services/CrossmintService";
+import { JsonWebTokenError, TokenExpiredError, verify } from "jsonwebtoken";
 
-// TODO: WAL-2563: this should resolve the user
-export async function verifyCrossmintSessionToken(
-    apiKey: string,
-    token: string,
-    environment: "production" | "staging"
-) {
-    const crossmintService = new CrossmintService(apiKey, token, environment);
+import { getCachedPublicKey, getFetchedPublicKey } from "./tokenAuth/publicKey";
 
-    let response;
+export async function verifyCrossmintSessionToken(apiKey: string, token: string) {
+    const crossmintService = new CrossmintService(apiKey, token);
     try {
-        response = await crossmintService.fetchCrossmintAPI({
-            endpoint: "/sdk/auth/me",
-            options: {
-                method: "POST",
-            },
-        });
+        return await verifyJWTWithPublicKey(crossmintService, "cached");
```

Comment:
this reads weird. Are you handling cache misses by catching an Error?

Reply:
hmm Im not following will ask you on DM about this